client/packages/common/src/ui/layout/tables/context/TableContext.tsx

```diff
@@ -61,12 +61,17 @@ export function useTableStore<T = TableStore>(
   selector?: (state: TableStore) => T,
   equalityFn?: (a: T, b: T) => boolean
 ): T {
-  if (!selector) return useStore(useContext(tableContext)) as unknown as T;
-
   const store = useContext(tableContext);
-  if (!equalityFn) return useStore(store, selector);
+  const storeWithoutSelector = useStore(store) as unknown as T;
+  const storeWithSelector = useStore(
+    store,
+    selector ?? ((_: TableStore) => storeWithoutSelector),
+    equalityFn
+  );
+
+  if (!selector) return storeWithoutSelector;
 
-  return useStore(store, selector, equalityFn);
+  return storeWithSelector;
```

Comment:
This was a bit tricky. Please check that its equivalent code.

Reply:
my concern with this is that it will call `useStore` with no parameters, which, according to the zustand docs doesn't sound desirable: `You can, but bear in mind that it will cause the component to update on every state change!`

Checking the number of renders, there's only a single render increase if the table contents are interacted with, so I think it's fine. Likely my testing is incomplete - I've gone through a few standard scenarios though and it seems fine.

I'd suggest this though - since the equality fn is optional:

```
export function useTableStore<T = TableStore>(
  selectorFn?: (state: TableStore) => T,
  equalityFn?: (a: T, b: T) => boolean
): T {
  const store = useContext(tableContext);
  const storeWithoutSelector = useStore(store) as unknown as T;
  const selector = selectorFn ?? ((_: TableStore) => storeWithoutSelector);

  return useStore(store, selector, equalityFn) as unknown as T;
}
```

Reply:
Thanks! update it